src/ipc/ipc4/helper.c

```diff
@@ -998,13 +999,13 @@ const struct comp_driver *ipc4_get_comp_drv(uint32_t module_id)
 #endif
 	}
 	/* Check already registered components */
-	drv = ipc4_get_drv(mod->uuid);
+	drv = ipc4_get_drv(&mod->uuid);
```

Comment:
currently `.uuid` in `struct sof_man_module` is an array, so yes, both these expressions generate the same result, and you probably want to change `.uuid` to a structure, which is why you need this change. But let's do that in that commit where you do it, not here.